lib/eslint.js

```diff
@@ -908,6 +908,8 @@ module.exports = (function() {
                 }
             }
 
+            ast.hasShebang = !!shebang;
```

Comment:
Yeah, this was what I was hoping to get some feedback on. The current behavior is to remove the shebang from the `comments` array and from the `leadingComments` of the first node in the Program body after parsing has completed. It also modifies the comment to look like a standard JS line comment. Now that we're calculating this on the fly, I need to figure out how `getComments()` can know which token represents a shebang comment (if one exists) so that it doesn't include it. Is there any possibility of adding a property to the token?